packages/pouch-storage/src/PouchCommandHistoryTable.ts

```diff
@@ -3,26 +3,258 @@ import {
+
+        this.cache.response.set(
+          this.cacheKey,
+          this.db
+            .allDocs({
+              include_docs: true,
+            })
+            .then(result => {
+              const allItems = result.rows
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                .map(row => row.doc!)
+                .filter(({ name }) => name);
+
+              log.debug(`Fetched ${allItems.length} command history items`);
+
+              const { toKeep, toPrune } = siftPrunableItems(
+                allItems,
+                COMMAND_HISTORY_ITEMS_MAX,
+                COMMAND_HISTORY_ITEMS_PRUNE
+              );
+
+              // If number of items in PouchDB has exceeded COMMAND_HISTORY_ITEMS_MAX
+              // prune them down so we have COMMAND_HISTORY_ITEMS_PRUNE left
+              if (toPrune.length) {
+                this.pruneItems(toPrune);
+              }
+
+              return {
+                docs: toKeep,
+              };
+            })
+        );
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const result = this.cache.response.get(this.cacheKey)!;
+
+      if (this.searchText == null || this.searchText === '') {
+        return result;
+      }
+
+      return {
+        ...result,
+        docs: (await result).docs.filter(({ name }) =>
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          name.toLowerCase().includes(this.searchText!)
+        ),
+      };
+    });
+  }
+
+  /**
+   * Override `PouchStorageTable.fetchInfo` so we can make use of
+   * `PouchCommandHistoryCache`
+   * @param selector
+   */
+  // Our current version of eslint + prettier doesn't always like the `override`
+  // keyword. Whenever we upgrade, we should annotate this function with `override`.
+  async fetchInfo(
+    selector: PouchDB.Find.Selector
+  ): Promise<
+    PouchDB.Find.FindResponse<CommandHistoryStorageItem & PouchStorageItem>
+  > {
+    return this.fetchData(selector);
+  }
+
+  /**
+   * Override `PouchStorageTable.fetchViewportData` to fetch command history
+   * storage items. This override provides caching + optimizations specific to
+   * command history.
+   *
+   * 1. The table doesn't support sorting, so ignore the sort param.
+   * 2. We cache the results to avoid excessive PouchDB queries during scrolling.
+   * The cache should only need to be invalidated when search filter changes or
+   * commands are added to the storage history.
+   *
+   * @param viewport
+   * @param selector
+   * @param _sort
+   * @returns Promise to array of command history storage items
+   */
+  // Our current version of eslint + prettier doesn't always like the `override`
+  // keyword. Whenever we upgrade, we should annotate this function with `override`.
+  async fetchViewportData(
+    viewport: StorageTableViewport,
+    selector: PouchDB.Find.Selector,
+    _sort: PouchDBSort
```

Comment:
One issue with not using the sort at all - we need the reverse of the table in the console input. Right now if you press `Up` in the console, it starts from the first command in your history, not the last one you entered. Need to make sure we at least handle that case.